Linking failure for Release builds

Now that tests build in the Release build, on clang++/MacOS, the build fails with a link error:
```
Undefined symbols for architecture arm64:
  "Module::Module<Nextsim::IOceanBoundary>::setExternalImplementation(std::__1::function<std::__1::unique_ptr<Nextsim::IOceanBoundary, std::__1::default_delete<Nextsim::IOceanBoundary>> ()>)", referenced from:
      Nextsim::DOCTEST_ANON_FUNC_15() in PrognosticData_test.cpp.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

--- a/scripts/module_builder.py
+++ b/scripts/module_builder.py
@@ -101,6 +101,8 @@ template <> std::unique_ptr<{strings[class_name]}> getInstance()
 {{
     addToConfiguredModules<{strings[class_name]}, {strings[module_class_name]}>();
 }}
+
+template class {module_templ};
 """)
     source.write("""
 } /* namespace Module */

